compiler/rustc_resolve/src/lib.rs

```diff
@@ -954,7 +951,10 @@ pub struct Resolver<'a> {
     arenas: &'a ResolverArenas<'a>,
     dummy_binding: &'a NameBinding<'a>,
 
-    crate_loader: CrateLoader<'a>,
+    local_crate_name: Symbol,
+    metadata_loader: Box<MetadataLoaderDyn>,
+    untracked: Untracked,
```

Comment:
I haven't read https://github.com/rust-lang/rust/pull/105462 yet, but introducing `Untracked` at this point and not when producing `ResolverOutputs` looks premature and seems to only lead to ugliness and downcasts in resolver code.
At least in this PR it looks like a regression.

Reply:
Unfortunately this is unavoidable if we want to make the stablehashcontext take a single reference instead of three, as the resolver needs to do stable hashing.

Any alternatives are highly appreciated, I just failed to come up with any.

Reply:
Ok, let's land as is (I don't currently have time to investigate, maybe later).